### PR TITLE
Adding documentation of client connection and data entry

### PR DESCRIPTION
**Pull request by maxhutch**
This is targeted a clearing a path to documenting link_by_uid,
but the scope of the documentation got a bit away from me.

### CODE REVIEW

File: docs/source/getting_started/connecting.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+======================================
+Connecting to the Citrine Platform
+======================================
+
+The Citrine Python Client connects to the Citrine Platform API.
+To create client, you must specify the specific site that you want to connect to and your API key:
```
Comment (bfolie) [inline docs/source/getting_started/connecting.rst]:
```suggestion
To create a client, you must specify the specific site that you want to connect to and your API key:
```

File: docs/source/data_entry.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
+.. data_entry:
+
+Data Entry
+=========================
+
+Creating and Data Model Objects
```
Comment (bfolie) [inline docs/source/data_entry.rst]:
```suggestion
Creating Data Model Objects
```

File: docs/source/data_entry.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+---------------------------------
+
+Each data object and template in the GEMD_ data model has a corresponding resource in the Citrine Python Client.
+For example, the :class:`citrine.resources.process_spec.ProcessSpec` class implements the ProcessSpec_ object in GEMD_.
+The Citrine Python Client implements are consistent with the GEMD_ model specification.
```
Comment (bfolie) [inline docs/source/data_entry.rst]:
```suggestion
The Citrine Python Client implementations are consistent with the GEMD_ model specification.
```

File: docs/source/data_entry.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+Identifiying Data Model Objects
+---------------------------------
+
+After registering a data model object, you will probably want to be able to find that object again.
+The easiest way to get an existing object is by one of its unique identifiers.
+Every data model object on the Citrine Platform has a platform-issued identifier, often referred to as the "Citrine Identifier" or "CitrineId".
+These identifiers are UUID4_, which are extremely robust but also not especially human readable.
+
+`Alternative identifiers`__ are an even easier way to recall data objects.
```
Comment (bfolie) [inline docs/source/data_entry.rst]:
```suggestion
`Alternative identifiers`__ are an easier way to recall data objects.
```

File: docs/source/data_entry.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+---------------------------------
+
+Data model objects are created on the Citrine Platform through the ``register`` method present in each data model object collection.
+For example:
+
+.. code-block:: python
+
+    project.process_specs.register(ProcessSpec(...))
+
+If you have taurus_ objects, e.g. :class:`taurus.entity.object.process_spec.ProcessSpec`, you can register it just like the objects defined in the Citrine Python Client.
```
Comment (bfolie) [inline docs/source/data_entry.rst]: Change taurus to GEMD?

File: docs/source/getting_started/basic_functionality.rst [hunk 6]
```diff
@@ -129,3 +108,51 @@ In order for the delete to be successful, the dataset must be empty.
...
+^^^^^^^^^^^^^^^^^^^^
+
+Starting with a specific root :class:`MaterialRun <citrine.resources.material_run.MaterialRun>`,
+you can retrieve the complete material history--every process, ingredient and material that went
+into the root material, as well as the measurements that were performed on all of those materials
+The method is :func:`~citrine.resources.material_run.MaterialRunCollection.get_history`,
+and it requires you to know a unique identifier (scope/id pair) for the material.
+
+LinkByUID
+^^^^^^^^^^
```
Comment (bfolie) [inline docs/source/getting_started/basic_functionality.rst]: I don't think this belongs under "Data Model Object Specific Methods"

Comment (maxhutch) [reply docs/source/getting_started/basic_functionality.rst]: Do you have a suggestion as to where to put it?

File: docs/source/data_entry.rst (hunk 5 above)
Comment (maxhutch) [reply docs/source/data_entry.rst]: @djack : how imminent is the taurus -> GEMD rename?

File: docs/source/getting_started/basic_functionality.rst (hunk 6 above)
Comment (bfolie) [reply docs/source/getting_started/basic_functionality.rst]: `getting_started/data_model`? It's not really a pice of functionality. I suppose it's not strictly part of the data model either, but is a fairly integral piece of its functioning

File: docs/source/getting_started/connecting.rst [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+======================================
+
+The Citrine Python Client connects to the Citrine Platform API.
+To create a client, you must specify the specific site that you want to connect to and your API key:
+
+.. code-block:: python
+
+    from citrine import Citrine
+    import os
+    citrine_client = Citrine(host="stage.citrine-platform.com", api_key=os.environ.get("CITRINE_API_KEY")
```
Comment (klennox-citrine) [inline docs/source/getting_started/connecting.rst]: In addition to setting the host and API key, I has to set port and and api scheme environment variables before I was able to connect. Per the confluence instruction:
export CITRINE_API_PORT=443
export CITRINE_API_SCHEME=https
I haven't seen either of these set using the Python interface.

Comment (maxhutch) [reply docs/source/getting_started/connecting.rst]: Do you have to?  Those have default values here, so I don't think they need to be set: https://github.com/CitrineInformatics/citrine-python/blob/develop/src/citrine/citrine.py#L14

**Comment by maxhutch** [comment]
> See my comment in re the "Connecting to the Citrine Platform" section. I think something somewhere goes looking for two additional environment variables (CITRINE_API_PORT and CITRINE_API_SCHEME), but there isn't any mention of them in the documentation.

Replied inline

File: docs/source/getting_started/connecting.rst (hunk 7 above)
Comment (klennox-citrine) [reply docs/source/getting_started/connecting.rst]: I can't replicate the original errors I remember getting, so I guess you don't.

**Pull request merged**